core/src/models/App.ts

```diff
@@ -207,6 +211,23 @@ export class App extends LoggedModel<App> {
     return { source, destination };
   }
 
+  async setConfigFilePath(newPath?: string) {
+    this.configFilePath = newPath ? newPath : `apps/${this.id}.json`;
+    await this.save();
+  }
+
+  async getConfigObject() {
```

Comment:
Every writable model now has a `getConfigObject` method that will build a writable config object.

As I dug deep into here, I realize we have _several_ N+1 problems throughout these methods where we do things like `getOptions()` when we could have already had the options eager loaded. To solve it was going to be a huge rewrite, so I kept the pattern the same here. I wasn't super worried about it in this context, primarily because we're not going to be working with a huge number of records at any given time, since we just writing configuration files.